nselib/http.lua

```diff
@@ -1571,7 +1571,11 @@ function parse_redirect(host, port, path, response)
   if ( not(u.host) ) then
     -- we're dealing with a relative url
     u.host = stdnse.get_hostname(host)
-    u.path = ((u.path:sub(1,1) == "/" and "" ) or "/" ) .. u.path -- ensuring leading slash
+    if ( u.path:sub(1,1) == "/") then
+      u.path = ((u.path:sub(1,1) == "/" and "" ) or "/" ) .. u.path -- ensuring leading slash
```

Comment:
Thanks a lot @nnposter and @cldrn for the comments. I have created a seperate PR for URL redirection issue : https://github.com/nmap/nmap/pull/918. Using url.absolute was giving me the correct path in every case. Besides, it also normalizes the path.